cli/medperf/comms/entity_resources/sources/synapse.py

```diff
@@ -0,0 +1,58 @@
+import synapseclient
+from synapseclient.core.exceptions import SynapseError
+from medperf.exceptions import (
+    CommunicationRetrievalError,
+    CommunicationAuthenticationError,
+)
+import os
+import shutil
+from .source import BaseSource
+import re
+
+
+class SynapseSource(BaseSource):
+    prefix = "synapse:"
+
+    @classmethod
+    def validate_resource(cls, value):
+        prefix = SynapseSource.prefix
+        if not value.startswith(prefix):
+            return
+
+        prefix_len = len(prefix)
+        value = value[prefix_len:]
+
+        if re.match(r"syn\d+$", value):
```

Comment:
What if no match? Is this an error that needs raised?

Reply:
The `validate_resource` methods either return the resource identifier or `None` in case of no match. The function `__parse_resource` in `utils.py` will raise an error if it doesn't receive any resource identifier when calling `validate_resource` from all supported sources

Reply:
Perhaps I should document the new code